### PR TITLE
feat: add biome vscode settings & use biome instead of eslint

### PR DESCRIPTION
**Pull request by bdbch**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "editor.defaultFormatter": "dbaeumer.vscode-eslint",
+  "editor.defaultFormatter": "biomejs.biome",
```
Comment (anbraten) [inline .vscode/settings.json:2]: Will this allow to still fix eslint errors on save as well?

Comment (bdbch) [reply .vscode/settings.json:2]: Nope - VSCode only allows one formatter to be set as a default I think.

Comment (bdbch) [reply .vscode/settings.json:2]: Is there an option to run biome via eslint? I think I remember there being an prettier plugin for eslint so maybe there's one for biome too?

**Comment by bdbch** [comment]
I tried to use the `eslint-config-biome` but it didn't work as expected. I think only using one formatting/linting library should be fine and since biome allows us to do both (except markdown parsing & linting/formatting) I'd suggest we use biome by default then.

File: package.json [hunk 2]
```diff
@@ -13,23 +13,16 @@
...
-    "eslint-plugin-import": "^2.25.2",
-    "eslint-plugin-n": "^15.0.0 || ^16.0.0 ",
-    "eslint-plugin-promise": "^6.0.0",
-    "nano-staged": "^0.8.0",
-    "typescript-eslint": "^8.16.0"
+    "nano-staged": "^0.8.0"
   },
   "nano-staged": {
-    "*.{js,ts,cjs,mjs}": ["biome check --write --"]
+    "*.{js,ts,cjs,mjs,json}": ["biome check --write --"]
```
Comment (anbraten) [inline package.json:26]: Would it support for markdown as well?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ jobs:
         run: pnpm -r lint
 
       - name: 📝 Check format
-        run: pnpm format:check
+        run: pnpm format
```
Comment (anbraten) [inline .github/workflows/ci.yml:38]: Does the normal format throw errors if something is not formatted?

File: package.json (hunk 2 above)
Comment (bdbch) [reply package.json:26]: https://github.com/biomejs/biome/issues/3718

I think they're still working on that.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bdbch) [reply .github/workflows/ci.yml:38]: Yep, this is the equivalent to `eslint` without `--fix`

**Pull request closed without merging**